lib/view_models/file_progress_provider.dart

```diff
@@ -24,8 +24,17 @@ class FileProgressProvider extends BaseModel {
 
   updateReceivedFileProgress(
       String transferId, FileTransferProgress fileTransferProgress) {
-    _receivedFileProgress[transferId] = fileTransferProgress;
-    notifyListeners();
+    double? prevPercent = _receivedFileProgress[transferId]?.percent;
+    double? newPercent = fileTransferProgress.percent;
+    double res = 1.0;
+    if (prevPercent != null && newPercent != null) {
+      res = newPercent - prevPercent;
+    }
+    if (res >= 1.0) {
+      fileTransferProgress.percent = fileTransferProgress.percent?.roundToDouble();
+      _receivedFileProgress[transferId] = fileTransferProgress;
+      notifyListeners();
```

Comment:
We can notify when percent change is more than one but we can store the value in `_receivedFileProgress` even when it's less than 1.

Reply:
If we do that, 
-> the difference between new value and previous value with never be more than one.
-> we will have to store the value of  lastNotifiedProgress to counter the first point.